### PR TITLE
Fonts rework

### LINKED ISSUE #11818: Use system-native font stack
Webfonts in general bring the issue that the page can not render completely without the font download finishing, which gives us one of the following issues:

- flash of invisible text (FOIT) which makes the browser render a invisible font until the download is done (equivalent to `font-display: block` and default behaviour in most browsers)
- flash of unstyled text (FOUT) which make the browser render a system font and then replace it with the downloaded font (equivalent to `font-display: swap`)

Both are bad in their own ways and I think we should eliminate all web fonts and just use a sytem-native font stack. This would match us with GitHub and GitLab which both also use native font stacks and do not rely on any web fonts.

GitHub has:
````
font-family: -apple-system,BlinkMacSystemFont,Segoe UI,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji;
````

GitLab has:
````
font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, "Noto Sans", Ubuntu, Cantarell, "Helvetica Neue", sans-serif, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol", "Noto Color Emoji";
````

### PR DESCRIPTION
**Pull request by silverwind**
- Use system fonts only for text to avoid FOUT
- Move font-awesome to npm/webpack
- Move NotoColorEmoji to web_src
- Remove presumably unneccesary 'PT Sans Narrow'
- Simplify webpack import exclusions

Fixes: https://github.com/go-gitea/gitea/issues/11818
Fixes: https://github.com/go-gitea/gitea/pull/11814

### CODE REVIEW

File: web_src/less/_base.less [hunk 1]
```diff
@@ -125,7 +75,7 @@
     .ui.steps .step .title,
     .ui.text.container,
     .ui.language > .menu > .item& {
-        font-family: Roboto, @fonts, sans-serif;
```
Comment (mrsdizzie) [inline web_src/less/_base.less:128]: Bless 🙏

File: web_src/less/_base.less [hunk 2]
```diff
@@ -60,19 +15,19 @@
     src:
         local('Noto Color Emoji'),
         local('Noto-Color-Emoji'),
-        url('../vendor/assets/noto-color-emoji/NotoColorEmoji.ttf') format('truetype');
+        url('../fonts/noto-color-emoji/NotoColorEmoji.ttf') format('truetype');
 }
 
-@default-fonts: -apple-system, BlinkMacSystemFont, system-ui, 'Segoe UI', Roboto, Helvetica, Arial, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol", "Noto Color Emoji";
+@default-fonts: -apple-system, BlinkMacSystemFont, system-ui, 'Segoe UI', Roboto, Helvetica, Arial, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol", "Noto Color Emoji" sans-serif;
```
Comment (silverwind) [inline web_src/less/_base.less:21]:
```suggestion
@default-fonts: -apple-system, BlinkMacSystemFont, system-ui, 'Segoe UI', Roboto, Helvetica, Arial, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol", "Noto Color Emoji", sans-serif;
```

**Review by mrsdizzie — APPROVED** [review]

**Comment by mrsdizzie** [comment]
LGTM  thanks! I'm also of the opinion that we don't really need to ship a emoji fallback webfont and the few people who run systems that don't install one by default and haven't already  installed one should expect to just not see them which must already be true for almost  every other site they visit.

Feels like an intentional decision for somebody not to have one of these fonts installed at this point and I can't imagine why they would prefer downloading a 9MB Web font instead of installing a real one system wide. I only agreed to it previously because I didn't want the emoji PR held back but think it is unnecessary for us to provide. Maybe for another PR though

**Comment by silverwind** [comment]
Yeah I think I tend to agree to remove the emoji fallback but as it stands, it does not hurt us to have. Browsers should only attempt the download if they can't display a certain emoji. Then there's the issue that the font does not work in Firefox because of a bug there, so it's only targeting Chrome/Edge on Linux really.

**Review by CirnoT — APPROVED** [review]
Looks good, would appreciate someone with Chinese locale testing it too however

**Review by lafriks — APPROVED** [review]

**Comment by silverwind** [comment]
> would appreciate someone with Chinese locale testing it too however

I did not change anything about the locale-specific font overrides, so I guess those will be fine.

**Comment by lunny** [comment]
> Looks good, would appreciate someone with Chinese locale testing it too however

I will test it after it deployed on try.gitea.io

**Pull request merged**